### PR TITLE
k2cli release path

### PR DESCRIPTION
**Pull request by coffeepac**
**What this PR does / why we need it**: adds release train to the k2cli jenkinsfile

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```allows for Jenkins based release automation
```

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ var keepAlive bool
 var logPath string
 var logSuccess bool
 var verbosity bool
+var k2Tag string
```
Comment (leahnp) [inline cmd/root.go]: Can you explain this variable? Do we need to add some documentation about it?

File: Jenkinsfile [hunk 2]
```diff
@@ -20,7 +27,7 @@ podTemplate(label: 'k2cli', containers: [
                 withEnv(["GOPATH=${WORKSPACE}/go/"]) {
                     stage('Test: Unit') {
                         kubesh 'cd go/src/github.com/samsung-cnct/k2cli/ && gosimple .'
-                        kubesh 'cd go/src/github.com/samsung-cnct/k2cli/ && make deps && make build'
+                        kubesh "cd go/src/github.com/samsung-cnct/k2cli/ && make deps && make build KLIB_VER=${k2_image_tag}"
```
Comment (leahnp) [inline Jenkinsfile]: Should we make the other 3 kubesh command use double quotes for uniformity?

**Comment by alejandroEsc** [comment]
CI may not pass, as there is an issue with unit tests which I address in a different PR, so I would suggest not letting CI block you.

File: cmd/root.go (hunk 1 above)
Comment (coffeepac) [inline cmd/root.go]: this is so we can compile a default k2 image version other then latest.  I will add a blurb in the 'cut a release' section for how to set this.

File: Jenkinsfile (hunk 2 above)
Comment (coffeepac) [inline Jenkinsfile]: yep

**Pull request merged**